### ISSUE TITLE
Factory settings in Azure Data Factory  |  Show billing report by pipeline

### ISSUE THREAD

**Issue by nmilosavljevic** [hashicorp/terraform-provider-azurerm#27314]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

After each Teraform configuration release, the Factory settings is changed from "by pipeline" to "by factory".
![Factory settings](https://github.com/user-attachments/assets/92e7578b-248c-4be2-8778-e05f0c3da70f)

How to configure Azure data factory settings to show billing "by pipeline", not "by factory" using Terraform?
Terraform Version: 3.116.0

### New or Affected Resource(s)/Data Source(s)

azurerm_data_factory

**Comment by gerrytan** [comment]
Problem can be reproduced, see: https://github.com/gerrytan/27314-datafactory-billing-report/blob/main/main.tf

ARM REST API to get/update billing configuration not yet implemented: https://learn.microsoft.com/en-us/rest/api/datafactory/factories/create-or-update?view=rest-datafactory-2018-06-01&tabs=HTTP

I will request this to be implemented first before support can be added to the Terraform resource.

**Comment by gerrytan** [comment]
Checked with @ruowan from Data Factory service team and billing configuration can be controlled via REST API by using following attribute in the request payload:

```
{
  "properties": {
    "globalConfigurations": {
      "PipelineBillingEnabled": "true"
    },
    ...
  },
}
```

This property is not yet documented on the REST API, hence it's missed on Terraform provider, however it's safe to be consumed via Terraform -- it's protected against breaking change.

Script to test factory GET / PUT via REST API: https://github.com/gerrytan/27314-datafactory-billing-report-azrest/blob/main/script.sh

I will create a provider improvement PR to add this support.

**Comment by gerrytan** [comment]
Upon checking the provider implementation, I cannot simply add `globalConfigurations` properties because the SDK model is generated from swagger. Refer to: https://github.com/hashicorp/terraform-provider-azurerm/blob/eb2d34534162d50ba71806fcded9110cc1cbbcfd/vendor/github.com/hashicorp/go-azure-sdk/resource-manager/datafactory/2018-06-01/factories/model_factoryproperties.go#L14

Adding this property to swagger might be considered a breaking change, and @ruowan informed me the ADF service team does not have any plan to release a new API version.

We might need to explore workaround using azapi provider until we get unblocked on the swagger spec.

**Comment by gerrytan** [comment]
@nmilosavljevic currently there is no plan to support billing configuration in azurerm provider, however you can use azapi to workaround this as per example below:

```
resource "azapi_resource" "adf" {
  type                      = "Microsoft.DataFactory/factories@2018-06-01"
  name                      = "my-adf"
  location                  = azurerm_resource_group.rg.location
  parent_id                 = azurerm_resource_group.rg.id
  schema_validation_enabled = false

  body = {
    properties = {
      globalConfigurations = {
        PipelineBillingEnabled = "true"
      }
    }
  }
}
```

Full example is available at https://github.com/gerrytan/27314-datafactory-billing-report/blob/bbcb497c631b87f410b4dc5d4d1b0c527d835ad2/main.tf.

See also documentation of `azapi_resource` : https://registry.terraform.io/providers/Azure/azapi/latest/docs/resources/resource

**Comment by katbyte** [comment]
This thread was left [with @gerrytan's conclusion](https://github.com/hashicorp/terraform-provider-azurerm/issues/27314#issuecomment-2558906521) and there has been no further activity in quite some time, so as part of a cleanup of historical issues we are closing it out.

If circumstances have changed and this affects you on the current version of the provider (v5.x), please open a new issue with fresh details referencing this one. Thank you!
